src/main/java/org/sagebionetworks/bridge/services/AccountWorkflowService.java

```diff
@@ -708,16 +710,29 @@ private boolean isRequestThrottled(ThrottleRequestType type, String userId) {
         // Generate key, which is in the form of channel-throttling:[type]:[userId].
         String cacheKey = "channel-throttling:" + type.toString().toLowerCase() + ":" + userId;
 
-        // We use Jedis to atomically increment the key value, and then set an expiration on it. The return value of
-        // exec() is the list of Jedis results in order. For the incr(), the result is the value after incrementing.
-        // This is the number of times we've requested in the last time period. If this exceeds the max requests, we're
-        // throttled.
-        int numRequests;
-        try (JedisTransaction transaction = jedisOps.getTransaction(cacheKey)) {
-            List<Object> resultList = transaction.incr(cacheKey).expire(cacheKey, channelThrottleTimeoutSeconds)
-                    .exec();
-            numRequests = ((Long) resultList.get(0)).intValue();
-        }
-        return numRequests > channelThrottleMaxRequests;
+        String numRequestsStr = jedisOps.get(cacheKey);
+        int numRequests = 0;
+        if (numRequestsStr != null) {
+            try {
+                numRequests = Integer.parseInt(numRequestsStr);
+            } catch (NumberFormatException ex) {
+                // Warn and fall back to 0.
+                LOG.warn("Found non-integer for cacheKey " + cacheKey + ": " + numRequestsStr);
+            }
+        }
+
+        if (numRequests < channelThrottleMaxRequests) {
+            // We've seen less than the maximum number of requests. We can let this request through without throttling.
+            // But we should increment the request count in Redis. Reset the expiration so that participants can't
+            // exceed the throttle limit by making a bunch of requests at the end of the throttle window.
+            jedisOps.setex(cacheKey, channelThrottleTimeoutSeconds, String.valueOf(numRequests + 1));
+            return false;
+        } else {
+            // We've seen at least as many requests as our limit (or more), so this next request will push us over the
+            // limit. Throttle this.
+            // Don't update Redis, since we throttle email/SMS sent, not requests. This allows the expiration to expire
+            // naturally.
+            return true;
+        }
```

Comment:
Doesn't need to happen here, but this throttling logic seems very useful to factor out so we can use it a number of places. In addition, I think it's fair to throw an exception that would return 429 status (too many requests) when the user hits the limit, so they understand what happened (and we see throttling in the logs). See BRIDGE-2415.